bundles/org.openhab.binding.ipcamera/README.md

```diff
@@ -0,0 +1,1312 @@
+
+There is also a Habpanel Widget worth checking out as the thread has an example on how to use the bindings mp4 filename history feature to track the filenames of recent recordings.
+<https://community.openhab.org/t/custom-widget-camera-history-and-live-popup/103082>
+
+**NOTE:** If you are using a tmpfs folder, you will need to ensure you do not run out of space. I use a rule to move the files out of the tmpfs as they are created.
+
+
+*.items
+
+```java
+Switch BackyardCamExternalMotion "External Motion" { channel="ipcamera:HIKVISION:BackyardCam:externalMotion" }
+String BackyardCamMp4Filename "Backyard Mp4 Filename" { channel="ipcamera:HIKVISION:BackyardCam:mp4Filename" }
+Number BackyardCamRecordMp4 "Backyard seconds to Record" { channel="ipcamera:HIKVISION:BackyardCam:recordMp4" }
+
+```
+
+*.rules
+
+```java
+
+rule "Record MP4 on External Zwave PIR"
+    when
+    Item BackyardCamExternalMotion changed to ON
+    then
+        BackyardCamMp4Filename.sendCommand( new DateTimeType().toString ) //create a file with the timestamp as the filename.
+        BackyardCamRecordMp4.sendCommand(5) // record 5 seconds    
+end
+
+rule "Do something when recording finished"
+    when
+    Item BackyardCamRecordMp4 changed to 0
+    then
+    var TimeStamp = BackyardCamMp4Filename.state
+    logInfo("camera.rules", "Mp4 recording "+ TimeStamp.toString +".mp4 is ready to be used.")
+end
+
+```
+
+
+## MJPEG Streams
+
+Cameras that have built in MJPEG abilities can stream to openHAB with the MJPEG format with next to no CPU load, less than 1 second lag, and FFmpeg does not need to be installed.
+Cameras without this ability can still use this binding to convert their cameras RTSP h264 format to mjpeg (keep reading for more on this below) and this will take a lot of CPU power to handle the conversion. 
+The alternative HLS format does not need the conversion and does not use any CPU to speak of, however due to HLS needing to buffer the files to disk, the HLS will result in lag (delay) behind real time.
+For video without a delay, you need mjpeg and without a camera that can create it, you will need to use a lot of CPU power. This could be done in a dedicated video server which will be the only way with lots of cameras unless you purchase cameras that have the ability built in.
+
+An alternative way to keep the CPU load low is to use the ``snapshots.mjpeg`` feature of the binding to create a stream from the cameras snapshots instead of the RTSP stream.
+This is limited to 1 frame a second but often results in far greater picture quality, so be sure to try the different ways and choose what you prefer.
+
+The main cameras that can do mjpeg with very low CPU load are Amcrest, Dahua, Hikvision, Foscam HD and Instar HD.
+To set this up, see the special setup steps for each brand in this readme.
```

Comment:
You could insert an anchor to this chapter.

Reply:
Done.